src/GraphQL/GraphQLBuilderExtensions.cs

```diff
@@ -1029,13 +1029,52 @@ public static IGraphQLBuilder AddValidationRule<TValidationRule>(this IGraphQLBu
         }
         #endregion
 
-        #region - AddMetrics -
+        #region - AddApolloTracing / AddMetrics -
+        /// <summary>
+        /// Registers <see cref="InstrumentFieldsMiddleware"/> within the dependency injection framework and
+        /// configures it to be installed within the schema, and configures responses to include Apollo
+        /// Tracing data when enabled via <see cref="ExecutionOptions.EnableMetrics"/>.
+        /// When <paramref name="enable"/> is <see langword="true"/>, configures execution to set
+        /// <see cref="ExecutionOptions.EnableMetrics"/> to <see langword="true"/>; otherwise leaves it unchanged.
+        /// </summary>
+        public static IGraphQLBuilder AddApolloTracing(this IGraphQLBuilder builder, bool enable = true)
+            => AddApolloTracing(builder, _ => enable);
+
+        /// <summary>
+        /// Registers <see cref="InstrumentFieldsMiddleware"/> within the dependency injection framework and
+        /// configures it to be installed within the schema, and configures responses to include Apollo
+        /// Tracing data when enabled via <see cref="ExecutionOptions.EnableMetrics"/>.
+        /// Configures execution to run <paramref name="enablePredicate"/> and when <see langword="true"/>, sets
+        /// <see cref="ExecutionOptions.EnableMetrics"/> to <see langword="true"/>; otherwise leaves it unchanged.
+        /// </summary>
+        public static IGraphQLBuilder AddApolloTracing(this IGraphQLBuilder builder, Func<ExecutionOptions, bool> enablePredicate)
+        {
+            if (enablePredicate == null)
+                throw new ArgumentNullException(nameof(enablePredicate));
+
+            builder.AddMiddleware<InstrumentFieldsMiddleware>();
+            builder.ConfigureExecution(async (options, next) =>
+            {
+                if (enablePredicate(options))
+                    options.EnableMetrics = true;
+                DateTime start = DateTime.UtcNow;
+                var ret = await next(options).ConfigureAwait(false);
+                if (options.EnableMetrics)
```

Comment:
```suggestion
                if (enabled)
```
What should be the behavior in case when some middleware set `EnableMetrics` to other value?

Reply:
I suggest to use the same calculated value inside this method to be consistent.

Reply:
No, it’s designed to allow for the user to set the EnableMetrics value anywhere else in the code, which may often be done when setting options from configuration settings before calling the document executer. In this manner it is consistent, and it is fully documented.

Reply:
OK. So tracing data **may** be calculated, but some middleware **after** actual document executing may set `EnableMetrics` to `false`, so there will be no tracing data in result. Right?

Reply:
That is a possible scenario, which makes sense.

```cs
services.AddGraphQL(b => b
    .AddApolloTracing() //note: order matters here
    .ConfigureExecution(async (opts, next) =>
    {
        var ret = await next(opts);
        // log captured metrics
        opts.EnableMetrics = false;
    }));
```

Reply:
👍